boards/google,veyron-jaq

```diff
@@ -30,6 +30,7 @@ assert_device_present dwc2-usb_otg-probed dwc2 ff580000.*
 
 assert_driver_present dwhdmi-rockchip-driver-present dwhdmi-rockchip
 assert_device_present dwhdmi-rockchip-probed dwhdmi-rockchip ff980000.*
+assert_driver_present dwhdmi-rockchip-driver-i2c-present dwhdmi-rockchip ff980000.hdmi/i2c-*
```

Comment:
Is this expected to pass with kernels older than v5.2?

If not then there should either be a kernel version check to test this driver conditionally.

Reply:
No, this will fail on older kernels. The commit was merged in 5.3-rc1.
```
$ git tag --contains=bf09924f21767e6bb7cb3aae48c48c2c2ab8261a
v5.3
v5.3-rc1
v5.3-rc2
v5.3-rc3
v5.3-rc4
v5.3-rc5
v5.3-rc6
v5.3-rc7
v5.3-rc8
v5.3.1
v5.3.2
v5.3.3
v5.3.4
v5.3.5
v5.4-rc1
v5.4-rc2
```
And this triggers the big question that I wanted to discuss.

If we care about older kernels or stable I think that more than check the tag we should check the commit hash. However, this will work for mainline but not for stable kernels where the same commit could be backported and have a different hash?

How we should handle this situation: A new feature is added to a driver so a new test is added to the test suite, but, of course, the test will fail on an old version of the driver. Thoughts?

There is any test suite that already does this?

Reply:
In all cases we should be able to just filter based on the kernel version (major.minor numbers):

1. if it's a fix that gets backported to all stable branches, then we may not need to have any filters as it'll always work once the patch has been applied
2. if the fix can't be backported to some older stable branches then we filter them (say, v4.9)
3. if it's not a fix but a change that won't get backported, then we can also filter based on the kernel version to run one check or another one depending on the version

For this patch `bf09924f2176 "ARM: dts: rockchip: Switch to builtin HDMI DDC bus on rk3288-veyron"` unless I'm mistaken I think we're in case 3 as it's not a fix so it's not being backported to any stable branches.  So if the kernel is v5.2 or older, check on i2c5.  Otherwise, check on hdmi i2c.

If you agree this should work in principle, then the next step is to add a utility to do this kernel version check in bootrr and then use it e.g. `if kernel_older_than "5.2"; ...`